### PR TITLE
can: Introduce can-primary alias

### PR DESCRIPTION
**Pull request by alexanderwachter**
This commit introduces the can-primary alias to identify
the primary CAN interface.
This alias is used for all samples and tests, so they don't
need to probe the right interface.

### CODE REVIEW

**Comment by alexanderwachter** [comment]
ping @chrta. Thanks for the inspiration ;-)

File: samples/drivers/CAN/src/main.c [hunk 1]
```diff
@@ -198,10 +198,7 @@ void main(void)
 	/* Usually the CAN device is either called CAN_0 or CAN_1, depending
```
Comment (chrta) [inline samples/drivers/CAN/src/main.c]: This documentation could be removed.

**Review by chrta — CHANGES_REQUESTED** [review]
Thanks for this change, i prefer this over the Kconfig solution.
Looks good, one minor comment.

File: samples/drivers/CAN/src/main.c (hunk 1 above)
Comment (alexanderwachter) [reply samples/drivers/CAN/src/main.c]: Thanks. Removed.

**Review by henrikbrixandersen — APPROVED** [review]

**Review by chrta — APPROVED** [review]

**Review by karstenkoenig — APPROVED** [review]
I think it is a really nice idea! Not sure how that relates/compares to something like the arduino_spi selection in the device tree files, are we sure this is not introducing a different solution to the same problem that was solved with the arduino headers?

**Comment by jfischer-phytec-iot** [comment]
An other workaround for the issue that we can not obtain a device by type. I thought about aliases for the sensors but that does not scale. Unlikely one would have many CAN devices but it does not work well for LEDs, sensors, serial devices, buttons...

**Comment by alexanderwachter** [comment]
> I think it is a really nice idea! Not sure how that relates/compares to something like the arduino_spi selection in the device tree files, are we sure this is not introducing a different solution to the same problem that was solved with the arduino headers?

Thats a different problem. Arduino has a fixed pin header and is implemented as kind of a device. In this PR we are just aliasing to mitigate the problem that nxp starts counting with zero and stm with one. The name primary (the next would be secondary) was chosen because it is somehow neutral.

**Comment by alexanderwachter** [comment]
> An other workaround for the issue that we can not obtain a device by type. I thought about aliases for the sensors but that does not scale. Unlikely one would have many CAN devices but it does not work well for LEDs, sensors, serial devices, buttons...

I think it is not a workaround. Sometimes you have more controllers (or redundancy or as a gateway)

**Review by erwango — APPROVED** [review]

**Comment by henrikbrixandersen** [comment]
Can all `DT_CAN_0_NAME` and `DT_CAN_1_NAME` dts fixups be removed at the same time?

**Comment by alexanderwachter** [comment]
@henrikbrixandersen not sure. Whats the benefit? Beside not needing the fixups any more.

**Comment by henrikbrixandersen** [comment]
> @henrikbrixandersen not sure. Whats the benefit? Beside not needing the fixups any more.

@alexanderwachter Clean up. No need to keep these fixups around if they are not used.

**Comment by alexanderwachter** [comment]
@henrikbrixandersen we can‘t ged rid of all fixups, because I‘m using them in the driver.
DT_CAN_1_NAME, for instance is used in the driver init to define the device name.
Any idea how to substitute them? The CAN_PRIMARY label is somehow misleading, because CAN_2 could also be the primary interface. I don‘t like the fixups either, but at the moment thats all we can do, I think.

**Review by jukkar — APPROVED** [review]

**Review by ydamigos — APPROVED** [review]

**Comment by alexanderwachter** [comment]
Rebased.

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**